### PR TITLE
Update Node.js to 16.18.0 and other WASM-related fixes

### LINKED ISSUE #2756: Follow-ups from WASM PR
https://github.com/unicode-org/icu4x/pull/2747

### LINKED ISSUE #2761: Fix up npm installation and caching on CI
We should never run `npm install` on CI, and in general we should avoid hitting npmjs.org in CI. We should leverage caching. We do this in general, but it doesn't seem to always work.

### PR DESCRIPTION
**Pull request by sffc**
Fixes #2761
Fixes #2756
Replaces #2762

### CODE REVIEW

File: .github/workflows/artifacts-build.yml [hunk 1]
```diff
@@ -137,18 +137,17 @@ jobs:
...
+        node-version: 16.18.0
         cache: 'npm'
         cache-dependency-path: '**/package-lock.json'
 
     - name: Init
       run: |
         cd ffi/diplomat/js/examples/wasm-demo
-        npm i
-        npm ci
+        npm ci --foreground-scripts
```
Comment (sffc) [inline .github/workflows/artifacts-build.yml:150]: `--foreground-scripts` makes npm forward the output of `build.sh`

**Comment by sffc** [comment]
The WASM demo builds with Node.js 16, without any `npm install` :smiley: 

https://storage.googleapis.com/icu4x-pr-artifacts/gha/a43f0b94218c34266e59d7c371509a6d0473d86c/wasm-demo/index.html

File: .github/workflows/artifacts-build.yml (hunk 1 above)
Comment (Manishearth) [reply .github/workflows/artifacts-build.yml:150]: nit: worth mentioning in a comment

File: .github/workflows/artifacts-build.yml [hunk 2]
```diff
@@ -137,18 +137,17 @@ jobs:
         service_account_key: ${{ secrets.ICU4X_GCP_SA_KEY }}
         export_default_credentials: true
 
-    - name: Install Node.js v14.17.0
+    - name: Install Node.js v16.18.0
```
Comment (Manishearth) [inline .github/workflows/artifacts-build.yml:140]: nit: CONTRIBUTING.md should also be bumped

**Review by Manishearth — APPROVED** [review]
feel free to land, minor followups requiested

**Pull request merged**